### DISCUSSION TITLE
Using clickhouse for storing kubernetes logs

### DISCUSSION THREAD

**Discussion by jatinarora0** [ClickHouse/ClickHouse#55274] [Q&A]
Hi everyone , i am trying to use clickhouse(database) metabase(visualisation) and vector(aggregation) for logging kubernetes data and application logs , what are the best practices to use clickhouse , as a best practice should we write data to a common table in clickhouse or we should use multiple tables for storing the data , if we should use multiple tables to store the data what are the best practices and steps to achieve that using vector.

this is the aggregator config i am using 
customConfig:
  data_dir: /vector-data-dir
  api:
    enabled: true
    address: 127.0.0.1:8686
    playground: false
  sources:
    vector:
      type: vector
      address: 0.0.0.0:6000
      type: vector
      version: "2"
  transforms:
    dots_to_underscores:
      type: remap
      inputs: [vector]
      source: |
        .kubernetes_namespace_labels = .kubernetes.namespace_labels
        .kubernetes_node_labels = .kubernetes.node_labels
        .kubernetes_pod_annotations = .kubernetes.pod_annotations
        .kubernetes_pod_labels = .kubernetes.pod_labels
        .kubernetes_container_image = .kubernetes.container_image
        .kubernetes_container_name = .kubernetes.container_name
        .kubernetes_pod_ip = .kubernetes.pod_ip
        .kubernetes_pod_ips = .kubernetes.pod_ips
        .kubernetes_pod_name = .kubernetes.pod_name
        .kubernetes_pod_namespace = .kubernetes.pod_namespace
        .kubernetes_pod_node_name = .kubernetes.pod_node_name
        .kubernetes_pod_owner = .kubernetes.pod_owner
        .kubernetes_pod_uid = .kubernetes.pod_uid
        del(.kubernetes)

  sinks:
    clickhouse:
      type: clickhouse
      inputs: [dots_to_underscores]
      database: vector
      endpoint: "http://ip:9090/"
      table: vector_logs
      compression: gzip
      auth:
        password: ''
        strategy: basic
        user: default
      batch:
        timeout_secs: 10
        max_events: 10000
        max_bytes: 104857600
      skip_unknown_fields: true
      buffer:
      
type: disk
      max_size: 1073741824 # 1GiB.
      when_full: block

**Comment by den-crane** [comment]
use a single table

**Reply by jatinarora0** [reply]
Thanks for the reply , I am using metabase for visualisation , when i query large amount of data it gives timeout on clickhouse node , any better way to manage large amount of data ?

**Reply by den-crane** [reply]
Did you try to query the same SQL using clickhouse-client? It could be a `metabase` issue during displaying a lot of datapoints.
